src/utils/slot.py

```diff
@@ -17,37 +18,108 @@ class NoSlotsAvailable(Exception):
 def get_first_non_missed_slot(
     cc: ConsensusClient,
     ref_slot: SlotNumber,
+    last_finalized_slot_number: SlotNumber,
```

Comment:
What if after ref_slot there is no finalized slots, only unfinalized?
And what to put to last_finalized_slot if we don't know it?

Reply:
A case when between `ref_slot` and `last_finalized_slot` no one finalized slot is determined as an exception.
We always know `last_finalized_slot` in each oracle cycle. It's equal `BaseModule._previous_finalized_slot_number`